src/wallet/wallet.h

```diff
@@ -677,25 +640,16 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
 
     isminetype IsMine(const CTxDestination& dest) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
     isminetype IsMine(const CScript& script) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
-    isminetype IsMine(const CTxIn& txin) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
     /**
      * Returns amount of debit if the input matches the
      * filter, otherwise returns 0
      */
     CAmount GetDebit(const CTxIn& txin, const isminefilter& filter) const;
     isminetype IsMine(const CTxOut& txout) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
-    CAmount GetCredit(const CTxOut& txout, const isminefilter& filter) const;
-    bool IsChange(const CTxOut& txout) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
-    bool IsChange(const CScript& script) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
-    CAmount GetChange(const CTxOut& txout) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
     bool IsMine(const CTransaction& tx) const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet);
     /** should probably be renamed to IsRelevantToMe */
     bool IsFromMe(const CTransaction& tx) const;
     CAmount GetDebit(const CTransaction& tx, const isminefilter& filter) const;
-    /** Returns whether all of the inputs match the filter */
-    bool IsAllFromMe(const CTransaction& tx, const isminefilter& filter) const;
-    CAmount GetCredit(const CTransaction& tx, const isminefilter& filter) const;
```

Comment:
How come you're able to move `GetCredit` but not `GetDebit`?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/22100#discussion_r642592067

> How come you're able to move `GetCredit` but not `GetDebit`?

I would like to move it, but it's kept in place for now to avoid circular dependencies. The problem is that the `CWallet::SyncTransaction` method needs to call `CWallet::AddToWalletIfInvolvingMe` which needs to call `CWallet::IsFromMe` which needs to call `CWallet::GetDebit`. These functions can't be moved out of `wallet.cpp` to `receive.cpp`, because `wallet.cpp` can't depend on `receive.cpp` (by design, higher level `receive.cpp` `spend.cpp` and a future `sync.cpp` are all meant to depend on the lower level `wallet.cpp` `CWallet` object).

I think a future fix for this will move `CWallet::SyncTransaction` along with chain attach and rescan code from `wallet.cpp` to `sync.cpp`. It should then be no problem for `sync.cpp` to depend on `receive.cpp` and for `CWallet::AddToWalletIfInvolvingMe` and related code to move to `receive.cpp`. The reason I didn't create `sync.cpp` and move these things already is that I was waiting for #20773 to split up the monolithic `CWallet::Create` function to make this easier.

Reply:
Or maybe in a (more compact) followup.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/22100#discussion_r642592067

>>>> How come you're able to move `GetCredit` but not `GetDebit`?

>>> I would like to move it, but it's kept in place for now to avoid circular dependencies.

>> Given that #20773 is merged, could the change to `GetDebit` be done now?

> Or maybe in a (more compact) followup.

Yes #20733 does make this easier, and yes this would make sense to do as a followup. Doing it in this PR wouldn't make great sense because this PR is not moving code just doing naming/declaration cleanups after a previous move. (Also this is not a tiny change, just in terms of number of lines). Will see what this looks like and post a followup.